### DISCUSSION TITLE
Zone rendering issue

### DISCUSSION THREAD

**Discussion by DotCat1985** [OrchardCMS/OrchardCore#11474] [Q&A]
Hi,
I developed a layout for a them as following:

```
@using OrchardCore.Themes.Services;
@using System.Globalization;
@inject ISiteThemeService SiteThemeService
@inject OrchardCore.Localization.ILocalizationService localizationService
@{
    var siteTheme = await SiteThemeService.GetSiteThemeAsync();
    var currentCultureName = Orchard.CultureName();
    var currentCultureNameShort = currentCultureName.Substring(0,2);
    var request = Orchard.HttpContext.Request;
}
<!DOCTYPE html>
<html lang="@currentCultureNameShort" dir="@Orchard.CultureDir()">
<head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0, shrink-to-fit=no" />
    <meta http-equiv="x-ua-compatible" content="ie=edge">
    <meta name="author" content="@siteTheme.Manifest.Author" />
    <meta name="copyright" content="@Site.SiteName" />
    <title>@await DisplayAsync(await New.PageTitle())</title>
    ...
    <resources type="HeadScript" />
    <resources type="HeadLink" />
    <resources type="Stylesheet" />
    @await RenderSectionAsync("HeadMeta", required: false)
    @await RenderSectionAsync("GdprBanner", required: false)
</head>
<body dir="@Orchard.CultureDir()">
    ...
    <main role="main">
        @await RenderBodyAsync()
    </main>
    ...
    <resources type="FootScript" />
</body>
</html>
```

where `GdprBanner` is a zone created with custom recipe and it contains the following Liquid snippet:

```
{% comment %}START IUBENDA BANNER{% endcomment %}
<script type="text/javascript">
    var _iub = _iub || [];
    ...
</script>
<script type="text/javascript" src="//cdn.iubenda.com/cs/ccpa/stub.js"></script>
<script type="text/javascript" src="//cdn.iubenda.com/cs/iubenda_cs.js" charset="UTF-8" async></script>
{% comment %}END IUBENDA BANNER{% endcomment %}
```

`@await RenderSectionAsync("GdprBanner", required: false)` always renders inside `<body>` instead in `<head>`.
Where should I move such code in order to render inside '<head>'?

**Comment by jtkech** [comment]
Yes, i noticed this behavior recently using net5.0, in my case I just tried it with the existing `HeadMeta` layout zone, same behavior by just using `@someVar`, it seemed that just using the `@` operator was breaking the rendered location, but the `<resources>` taghelpers (without any `@`) was rendered a the right location so I thought that we would need a `RenderSectionAsync` taghelper.

Didn't try it but maybe it was already working with Liquid tags as used in TheBlogTheme, you could try it.

But just re-tried with the last dev, so using net6.0, and it works, that said it also renders extra widget html div which is not good in the page Head section. So anyway you have better to use a Liquid widget in any other layout zone, e.g. the Head layout zone (not the page Head section) and then in your widget use Liquid Tags, those dedicated for injecting resources at the right locations.

    {% script src:"~/Some.Theme/js/SomeScript.js", at: "Head" %}

With more optional parameters

    {% script src:"~/Some.Theme/js/SomeScript.js", debug_src:"~/Some.Theme/js/SomeScript.js", depends_on:"dep1, dep2", at: "Head" %}

Or if all defined in a ResourceManifestOptionsConfiguration.cs, just specify the resource name.

    {% script name:"MyResourceName", at: "Head" %}
